airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/BigQueryDestinationTest.java

```diff
@@ -235,6 +237,40 @@ void testCheckFailure(final DatasetIdResetter resetDatasetId) {
     assertEquals(expected, actual.withMessage(""));
   }
 
+  @ParameterizedTest
+  @MethodSource("datasetIdResetterProvider")
+  void testCheckFailureInsufficientPermissionForCreateDataset(final DatasetIdResetter resetDatasetId) throws IOException {
+
+    if (!Files.exists(CREDENTIALS_WITHMISSED_CREATE_DATASET_ROLE_PATH)) {
```

Comment:
I see this occurring in other parts of the code, what exactly is this supposed to do that is causing a test to throw an `IllegalStateException`?

Reply:
This it a test creds that supposed to fail.
I will fail on CI if somebody removes or change secret in GSM or to somebody, who will try to run integrationTest locally without having these creds locally in secret folder. This is a common integration test's approach for connectors that calls real DB  i.e. requires real creds that obviously can't be committed.